src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -252,6 +252,25 @@ public void checkCourseAndExamAccessForTeachingAssistantElseThrow(Long courseId,
         checkExamBelongsToCourseElseThrow(courseId, examId);
     }
 
+    /**
+     * Checks if the current user is has a student exam in the given exam of the given course and that the exam
+     * belongs to the given course.
+     *
+     * @param courseId The id of the course
+     * @param examId   The id of the exam
+     */
+    public void checkCourseAndExamAccessForStudentElseThrow(Long courseId, Long examId) {
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, currentUser);
+        if (authorizationCheckService.isAtLeastInstructorInCourse(course, currentUser)) {
+            checkExamBelongsToCourseElseThrow(courseId, examId);
+        }
+        else if (!studentExamRepository.existsByExam_CourseIdAndExamIdAndUserId(courseId, examId, currentUser.getId())) {
```

Comment:
unnecessary else
```suggestion
        if (!studentExamRepository.existsByExam_CourseIdAndExamIdAndUserId(courseId, examId, currentUser.getId())) {
```

Reply:
We don't need to (also shouldn't) check `existsByExam_CourseIdAndExamIdAndUserId` when the current user is at least instructor so I think the `else` is needed or am I missing something maybe?